presto-prometheus/src/test/java/io/prestosql/plugin/prometheus/TestPrometheusSplit.java

```diff
@@ -0,0 +1,543 @@
+    private final PrometheusSplit split = new PrometheusSplit(URI.create("http://127.0.0.1/test.file"));
+    private static final Metadata METADATA = createTestMetadataManager();
+    public static final TypeManager TYPE_MANAGER = new InternalTypeManager(METADATA);
+    static final int NUMBER_MORE_THAN_EXPECTED_NUMBER_SPLITS = 100;
+    PrometheusTableHandle prometheusTableHandle;
+
+    @BeforeClass
+    public void setUp()
+            throws Exception
+    {
+        prometheusHttpServer = new PrometheusHttpServer();
+    }
+
+    @BeforeMethod
+    public void resetTableHandle()
+            throws Exception
+    {
+        prometheusTableHandle = new PrometheusTableHandle("schemaName", "tableName");
+    }
+
+    @Test
+    public void testAddresses()
+    {
+        // http split with default port
+        PrometheusSplit httpSplit = new PrometheusSplit(URI.create("http://prometheus.com/prometheus"));
+        assertEquals(httpSplit.getAddresses(), ImmutableList.of(HostAddress.fromString("prometheus.com")));
+        assertEquals(httpSplit.isRemotelyAccessible(), true);
+
+        // http split with custom port
+        httpSplit = new PrometheusSplit(URI.create("http://prometheus.com:8080/prometheus"));
+        assertEquals(httpSplit.getAddresses(), ImmutableList.of(HostAddress.fromParts("prometheus.com", 8080)));
+        assertEquals(httpSplit.isRemotelyAccessible(), true);
+
+        // http split with default port
+        PrometheusSplit httpsSplit = new PrometheusSplit(URI.create("https://prometheus.com/prometheus"));
+        assertEquals(httpsSplit.getAddresses(), ImmutableList.of(HostAddress.fromString("prometheus.com")));
+        assertEquals(httpsSplit.isRemotelyAccessible(), true);
+
+        // http split with custom port
+        httpsSplit = new PrometheusSplit(URI.create("https://prometheus.com:8443/prometheus"));
+        assertEquals(httpsSplit.getAddresses(), ImmutableList.of(HostAddress.fromParts("prometheus.com", 8443)));
+        assertEquals(httpsSplit.isRemotelyAccessible(), true);
+    }
+
+    @Test
+    public void testJsonRoundTrip()
+    {
+        JsonCodec<PrometheusSplit> codec = jsonCodec(PrometheusSplit.class);
+        String json = codec.toJson(split);
+        PrometheusSplit copy = codec.fromJson(json);
+        assertEquals(copy.getUri(), split.getUri());
+
+        assertEquals(copy.getAddresses(), ImmutableList.of(HostAddress.fromString("127.0.0.1")));
+        assertEquals(copy.isRemotelyAccessible(), true);
+    }
+
+    @Test
+    public void testQueryWithTableNameNeedingURLEncodeInSplits()
+            throws URISyntaxException
+    {
+        PrometheusConnectorConfig config = new PrometheusConnectorConfig();
+        LocalDateTime now = LocalDateTime.of(2019, 10, 2, 7, 26, 56, 00);
+        PrometheusTimeMachine.useFixedClockAt(now);
+
+        dataUri = prometheusHttpServer.resolve("/prometheus-data/prom-metrics-non-standard-name.json");
+        config.setPrometheusURI(dataUri);
```

Comment:
I would recommend extracting common code so that we can focus the actual test.

Reply:
I used to share that opinion about test code. But I've read a few articles that advocate making test code use lots of redundant copy/paste code. The idea being: when the test fails, the dev troubleshooting the failure can just read everything in one place, as much as possible, and within reason. Since I've adopted this plan, I have found as the troubleshooting dev, I sure like it a lot better than traipsing through some beautifully DRY test code to learn all the structures I need to understand the test failure.

I'm okay with DRYing this test code out if really desired, let me know.

Reply:
Yes, we sometimes break DRY in tests as you mentioned, but I suppose the current TestPrometheusSplit is too redundant. You can find existing tests have helper methods to reduce redundant code (e.g. TestSubqueries).

Reply:
Fair enough. I DRYed it out a bit with a method covering the common config items. In db232cad